File: tests/src/org.apache.kafka/kafka-clients/4.2.0/src/test/java/org_apache_kafka/kafka_clients/GeneratedMessageV3Test.java
/*
 * Copyright and related rights waived via CC0
 *
 * You should have received a copy of the CC0 legalcode along with this
 * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
 */
package org_apache_kafka.kafka_clients;

import org.apache.kafka.shaded.com.google.protobuf.Any;
import org.apache.kafka.shaded.com.google.protobuf.Descriptors;
import org.apache.kafka.shaded.com.google.protobuf.StringValue;
import org.junit.jupiter.api.Test;

import static org.assertj.core.api.Assertions.assertThat;

public class GeneratedMessageV3Test {

    @Test
    void packsParsesUnpacksAndPrintsGeneratedProtobufMessage() throws Exception {
        StringValue value = StringValue.of("payload");
        Any packed = Any.pack(value);
        byte[] bytes = packed.toByteArray();

        Any parsed = Any.parseFrom(bytes);
        StringValue unpacked = parsed.unpack(StringValue.class);
        Descriptors.Descriptor descriptor = StringValue.getDescriptor();

        assertThat(unpacked.getValue()).isEqualTo("payload");
        assertThat(parsed.toString()).contains("type.googleapis.com");
        assertThat(descriptor.findFieldByName("value")).isNotNull();
    }
}

File: tests/src/org.apache.pekko/pekko-protobuf-v3_2.13/1.1.5/src/test/scala/org_apache_pekko/pekko_protobuf_v3_2_13/GeneratedMessageV3Test.scala
/*
 * Copyright and related rights waived via CC0
 *
 * You should have received a copy of the CC0 legalcode along with this
 * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
 */
package org_apache_pekko.pekko_protobuf_v3_2_13

import org.apache.pekko.protobufv3.internal.Descriptors
import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test

class GeneratedMessageV3Test {
  @Test
  def generatedMessageV3FieldAccessorsUseGeneratedAccessMethods(): Unit = {
    val field: Descriptors.FieldDescriptor = GeneratedMessageV3Probe.getDefaultInstance
      .getDescriptorForType
      .findFieldByName("quantity")
    val message: GeneratedMessageV3Probe = GeneratedMessageV3Probe.newBuilder()
      .setQuantity(7)
      .build()

    assertThat(message.hasField(field)).isTrue()
    assertThat(message.getField(field)).isEqualTo(7)

    val builder: GeneratedMessageV3Probe.Builder = message.toBuilder
    builder.setField(field, 11)

    assertThat(builder.hasField(field)).isTrue()
    assertThat(builder.getField(field)).isEqualTo(11)

    builder.clearField(field)

    assertThat(builder.hasField(field)).isFalse()
    assertThat(builder.build().getQuantity).isEqualTo(0)
  }
}
